WebView/app/src/main/java/com/android/samples/webviewdemo/MainActivity.kt

```diff
@@ -128,6 +133,27 @@ class MainActivity : AppCompatActivity() {
         val jsObjName = "jsObject"
         val allowedOriginRules = setOf<String>("https://gcoleman799.github.io")
 
+        // FORCE DARK
+        // Check if the system is set to light or dark mode
+        val nightModeFlag = resources.configuration.uiMode and Configuration.UI_MODE_NIGHT_MASK
+        if (nightModeFlag == Configuration.UI_MODE_NIGHT_YES) {
+            // Switch WebView to dark mode; uses default dark theme
+            if (WebViewFeature.isFeatureSupported(WebViewFeature.FORCE_DARK)) {
+                WebSettingsCompat.setForceDark(
+                    binding.webview.settings,
+                    WebSettingsCompat.FORCE_DARK_ON
+                )
+            }
```

Comment:
I was thinking about this and I realized I'm not sure how appcompat's day/night works. If the user switches modes and switches back to the app (without killing it), does the app get re-themed?

The chart in https://developer.android.com/guide/components/activities/activity-lifecycle might provide some context for my question.

Not something we need to fix in this PR, but we should eventually verify WebView is getting themed consistently with appcompat day/night.

Reply:
If I am understanding lifecycles correctly then I believe that when the user switches modes without killing it then the app does get re-themed.

Reply:
As Grace wrote, yes, it does, since the change from day<->night mode is a configuration change the activity will be restarted (and pickup the new values to apply the theme).

Reply:
Does that mean `onCreate` is called again, or just `onResume`? If it's just `onResume`, then we need to move this code to `onResume` for consistency.

Reply:
Sorry, not "restarted" but "recreated". So the old Activity is destroyed and a new Activity is created. So it goes through `onCreate`, `onStart`, etc...